server/src/weaverbird/backends/pypika_translator/translators/athena.py

```diff
@@ -10,5 +18,41 @@ class AthenaTranslator(SQLTranslator):
     SUPPORT_ROW_NUMBER = True
     SUPPORT_SPLIT_PART = True
 
+    # Since Athena doesn't handle date comparaison as string and raise :
+    # awswrangler.exceptions.QueryFailed: SYNTAX_ERROR: line 31:27: '<=' cannot be applied to date, varchar(25).
+    # You may need to manually clean the data at location 's3://to....' before retrying.
+    def filter(
+        self,
+        *,
+        builder: 'QueryBuilder',
+        prev_step_name: str,
+        columns: list[str],
+        step: "FilterStep",
+    ) -> StepContext:
+
+        # To handle relative date formats
+        cond = step.condition
+        if isinstance(cond, (ConditionComboAnd, ConditionComboOr)):
+
+            operator = 'and_' if isinstance(cond, ConditionComboAnd) else 'or_'
+
+            for i, sub_cond in enumerate(getattr(cond, operator)):
+                if isinstance(sub_cond, DateBoundCondition):
+                    value = getattr(sub_cond, 'value')
+                    if isinstance(value, RelativeDate):
+                        value = evaluate_relative_date(value).strftime("%Y-%m-%d")
+                    elif isinstance(value, datetime):
+                        value = value.strftime("%Y-%m-%d")
```

Comment:
Spotted something when testing athena, if i add to the strftime
'%H:%M:%S', i got this error :
awswrangler.exceptions.QueryFailed: INVALID_CAST_ARGUMENT:
Value cannot be cast to date: 2022-01-05 00:00:00.
You may need to manually clean the data at location
's3://-------------------------------/output/tables/-----'
before retrying. Athena will not delete data in your account.

Does that mean the dataset should be updated ? 🤔